### PR TITLE
Update runs/search with 4 new fields Refs. #1928

### PR DESCRIPTION
**Pull request by atodorov**
@AirinB the previous PR got closed because I've deleted the target branch. I've rebased the source branch towards the latest master and opened this PR instead.

### CODE REVIEW

File: tcms/testruns/templates/testruns/search.html [hunk 1]
```diff
@@ -71,6 +71,28 @@
                 <input class="bootstrap-switch" id="id_running" name="running" type="checkbox" {% if form.running.value != "0" %}checked{% endif %} data-on-text="{% trans 'Yes' %}" data-off-text="{% trans 'No' %}">
             </div>
         </div>
+        <div class="form-group">
+            <label class="col-md-1 col-lg-1" for="id_start_date">{% trans "Start Date" %}</label>
```
Comment (atodorov) [inline tcms/testruns/templates/testruns/search.html:75]: Don't capitalize field names -> Start Date -> Start date

File: tcms/testruns/templates/testruns/search.html [hunk 2]
```diff
@@ -71,6 +71,28 @@
...
+            <div class="col-md-3 col-lg-3">
+                <input id="id_start_date" type="datetime-local" class="form-control">
```
Comment (atodorov) [inline tcms/testruns/templates/testruns/search.html:77]: These fields don't open a calendar widget, see plan/search page for example.

File: tcms/testruns/templates/testruns/search.html [hunk 3]
```diff
@@ -71,6 +71,28 @@
...
+            </div>
+
+            <label class="col-md-1 col-lg-1" for="id_planned_stop">{% trans "Planned Stop" %}</label>
+            <div class="col-md-3 col-lg-3">
+                <input id="id_planned_start" type="datetime-local" class="form-control">
+            </div>
+        </div>
 
         <div class="form-group">
             <label class="col-md-1 col-lg-1" for="id_tag">{% trans "Tag" %}</label>
```
Comment (atodorov) [inline tcms/testruns/templates/testruns/search.html:98]: Move the Tag field after Plan ID (top-right) to use the available free space on screen and bring the results table upwards a little bit.

File: tcms/testruns/templates/testruns/search.html [hunk 4]
```diff
@@ -71,6 +71,28 @@
...
+        </div>
+        <div class="form-group">
+            <label class="col-md-1 col-lg-1" for="id_planned_start">{% trans "Planned Start" %}</label>
+            <div class="col-md-3 col-lg-3">
+                <input id="id_planned_start" type="datetime-local" class="form-control">
+            </div>
+
+            <label class="col-md-1 col-lg-1" for="id_planned_stop">{% trans "Planned Stop" %}</label>
+            <div class="col-md-3 col-lg-3">
+                <input id="id_planned_start" type="datetime-local" class="form-control">
```
Comment (atodorov) [inline tcms/testruns/templates/testruns/search.html:93]: All of the 4 fields should be tied to the search functionality on this page in order to be useful.

**Review by atodorov — COMMENTED** [review]
Added my previous comments from #2267

**Comment by AirinB** [comment]
@atodorov I added the before and after fields but what I don't understand is how I should send it or maybe display them in another way to be able to send just 1 field for each of the 4 ones (start date, stop date , planned stop and planned start) Also when I send the form I see that the format os not accepted. Could you please advise me regarding this?

**Review by atodorov — COMMENTED** [review]
Rebase to latest master

File: tcms/testruns/static/testruns/js/search.js [hunk 5]
```diff
@@ -28,6 +51,38 @@ $(document).ready(function() {
...
+                params['before_planned_start'] = $('#id_before_planned_start').val();
+            }
+
+            if ($('#id_after_planned_stop').val()) {
+                params['after_planned_stop'] = $('#id_after_planned_stop').val();
+            }
+
+            if ($('#id_before_planned_stop').val()) {
+                params['before_planned_stop'] = $('#id_before_planned_stop').val();
+            }
```
Comment (atodorov) [inline tcms/testruns/static/testruns/js/search.js:84]: https://docs.djangoproject.com/en/3.1/topics/db/queries/#field-lookups

**Review by atodorov — COMMENTED** [review]
Some changes requested.

**Comment by atodorov** [comment]
> @atodorov I added the before and after fields but what I don't understand is how I should send it or maybe display them in 

For the display part see:
https://github.com/kiwitcms/Kiwi/commit/a92857e2f9815c20e716c36d9e129fcf082b7db5

**Comment by AirinB** [comment]
@atodorov I will squash the commits a little bit later, I have some trouble doing this 😄 . But let me know if you have some feedback regarding the functionality.

File: tcms/testruns/static/testruns/js/search.js [hunk 6]
```diff
@@ -70,8 +102,8 @@ $(document).ready(function() {
                 params['manager__username__startswith'] = $('#id_manager').val();
             };
 
-            if ($('#id_default_tester').val()) {
-                params['default_tester__username__startswith'] = $('#id_default_tester').val();
+      if ($('#id_default_tester').val()) {
+        params['default_tester__username__startswith'] = $('#id_default_tester').val();
```
Comment (atodorov) [inline tcms/testruns/static/testruns/js/search.js:106]: Unrelated formatting change, revert it back.

File: tcms/testruns/static/testruns/js/search.js [hunk 7]
```diff
@@ -90,7 +122,7 @@ $(document).ready(function() {
                         result += '<p class="help-block">' + data.stop_date + '</p>';
                     }
                     return result;
-                }
+        }
```
Comment (atodorov) [inline tcms/testruns/static/testruns/js/search.js:125]: unrelated formatting change, revert it.  Also actually breaks the existing formatting.

File: tcms/testruns/templates/testruns/search.html [hunk 8]
```diff
@@ -74,6 +74,99 @@
                 <input class="bootstrap-switch" id="id_running" name="running" type="checkbox" {% if form.running.value != "0" %}checked{% endif %} data-on-text="{% trans 'Yes' %}" data-off-text="{% trans 'No' %}">
             </div>
         </div>
+{#    TODO#}
+
+<!-- playing around with making these fields occupy less space on the screen -->
```
Comment (atodorov) [inline tcms/testruns/templates/testruns/search.html:79]: Remove the TODO and the unrelated comments.

**Review by atodorov — COMMENTED** [review]
Some changes requested.

I suspect there may be conflicts since the HTML page I think was recently updated, see e71c4a32f2f4336564d3ccd7683e376622b67844. A rebase should solve the issues.

File: tcms/testruns/templates/testruns/search.html [hunk 9]
```diff
@@ -114,11 +205,15 @@
...
+    <script src="{% static 'moment/min/moment-with-locales.min.js' %}"></script>
+    <script src="{% static 'eonasdan-bootstrap-datetimepicker/build/js/bootstrap-datetimepicker.min.js' %}"></script>
+    <script src="{% static 'bootstrap-switch/dist/js/bootstrap-switch.min.js' %}"></script>
+    <script src="{% static 'bootstrap-select/dist/js/bootstrap-select.min.js' %}"></script>
+
+    <script src="{% static 'js/jsonrpc.js' %}"></script>
+    <script src="{% static 'js/utils.js' %}"></script>
+    <script src="{% static 'js/pagination.js' %}"></script>
+    <script src="{% static 'testruns/js/search.js' %}"></script>
+
```
Comment (atodorov) [inline tcms/testruns/templates/testruns/search.html:217]: Please don't introduce unrelated changes, especially when changing the order of the lines. This makes it very hard to figure out if something essential has been changed or not. And if it has been then answering the question: is there something missing or everything looks fine. 

If nothing is missing and all looks fine then there's very little reason for such changes in the first place other than eye candy. If such is the case then formatting changes need to be introduced separately of functional related changes because of the above mentioned reasons.

**Review by atodorov — COMMENTED** [review]
Few minor issues. 

I'm unable to test anymore because when I rebase it looks like one of the commits goes away and I'm losing half the fields on the page. Not sure what's going on but I'll investigate it once you make the final commit. 

Note: make a new commit so that I can see something is different, not sure what's going on ATM.

**Comment by AirinB** [comment]
@atodorov hope it is better now

**Comment by atodorov** [comment]
> @atodorov hope it is better now

Much better thanks. I now see the 2 new javascript files added to the page. 

I'll dig into the rebase issue next.

**Comment by AirinB** [comment]
@atodorov hope this looks okay :D
Screen Shot 2021-03-19 at 10 56 51 AM
